core/src/edu/asu/c3simulator/widgets/NavigationPanel.java

```diff
@@ -113,28 +146,12 @@ public void addButton(String buttonName, Screen targetedScreen)
 	 *            The screen that the button maps to. If null, no actionListener will be
 	 *            added to the button.
 	 */
-	public boolean addSubButton(String parentButtonName, String buttonName,
+	public void addSubButton(String parentButtonName, String buttonName,
 			Screen targetedScreen)
 	{
-		String storedParentButtonName = storedButtonName(parentButtonName);
-		if (!storedParentButtonName.equalsIgnoreCase(""))
-		{
-			if (subButtons.containsKey(storedParentButtonName))
-			{
-				LinkedHashMap<String, Screen> relativeSubButtons = subButtons
-						.get(storedParentButtonName);
-				relativeSubButtons.put(buttonName, targetedScreen);
-			}
-			else
-			{
-				LinkedHashMap<String, Screen> newSubButtons = new LinkedHashMap<String, Screen>();
-				newSubButtons.put(buttonName, targetedScreen);
-				subButtons.put(storedParentButtonName, newSubButtons);
-			}
-			recreateNavigationPanel();
-			return true;
-		}
-		return false;
+		ButtonNodeListener parent = buttons.get(parentButtonName);
```

Comment:
According to [this](http://docs.oracle.com/javase/7/docs/api/java/util/HashMap.html#get(java.lang.Object)), it uses key.equals().

However, Java Strings are immutable, and optimized to map to an equivalent String instance if possible. That is, if two strings are declared like this:
    String a = "test";
    String b = "test";
Then a==b will return true, as will a.equals(b). 
If a new instance was created as follows:
    String a = "test";
    String b = new String("test");
then a==b would return false, but a.equals(b) would return true.